docs/reference/TensorFlow.md

```diff
@@ -1,405 +1,394 @@
+    ```
 
-The code samples audio from the microphone on the K66F. The audio is run through a Fast Fourier transform to create a spectrogram. The spectrogram is then fed into a pre-trained machine learning model. The model uses a [convolutional neural network](https://developers.google.com/machine-learning/practica/image-classification/convolutional-neural-networks) to identify whether the sample represents either the command “yes” or “no”, silence, or an unknown input. We will explore how this works in more detail later in the guide.
+    <!--The install page for Mbed CLI now requires 3.7. Why are we asking for 2.7?-->
+    <!--need to explain what `config root .` and `deploy` do and why we need them - neither one is part of a standard workflow where you use Mbed CLI to import an application, so this is a special case-->
+    <!--and why are we compiling here? We compile again two steps down, with the flash parameter-->
 
-Here are descriptions of some interesting source files:
+    For some compilers<!--we only support two, and you specifically asked for GNU, so in what case will this happen?-->, you may get a compilation error in `mbed_rtc_time.cpp`. Go to  `mbed-os/platform/mbed_rtc_time.h`  and comment out line 32 and line 37:
 
--   nxp_k66f[/audio_provider.cc](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/micro/examples/micro_speech/nxp_k66f/audio_provider.cc) captures audio from the microphone on the device.
+    ```
+    //#if !defined(__GNUC__) || defined(__CC_ARM) || defined(__clang__) struct timeval {
+    time_t  tv_sec; int32_t tv_usec;
+    };
+    //#endif
+    ```
 
--   [micro_features/micro_features_generator.cc](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/micro/examples/micro_speech/micro_features/micro_features_generator.cc): uses a Fast Fourier transform to create a spectrogram from audio.
+1. Connect your board to your computer over USB. On your board, when the Ethernet port is facing you, the micro USB port is to its left.
 
--   [micro_features/model.cc](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/micro/examples/micro_speech/micro_features/model.cc). This file is the machine learning model itself, represented by a large array of unsigned char values.
+    Your board appears as storage on your computer. If your system does not recognize the board with the `mbed detect` command, follow the instructions for setting up [DAPLink](https://armmbed.github.io/DAPLink/?board=FRDM-K66F).
 
--   [command_responder.cc](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/micro/examples/micro_speech/command_responder.cc) is called every time a potential command has been identified.
+1. Flash the application to the board:
+    <!--I compiled two steps ago - that probably needs to be removed-->
+    ```
+    mbed compile -m K66F -t GCC_ARM –flash
+    ```
+1. Deploy the example to your K66F<!--but we flshed already. What are we doing here?-->
```

Comment:
Again, following the format in the original [guide on developer.arm.com](https://developer.arm.com/solutions/machine-learning-on-arm/developer-material/how-to-guides/build-arm-cortex-m-voice-assistant-with-google-tensorflow-lite/single-page). They had two ways of doing things. One where we compile and flash on the same command `mbed compile -m K66F -t GCC_ARM –flash` and one where we just compile then copy the binary over to the board to flash it, as shown below with `cp ./BUILD/K66F/GCC_ARM/mbed.bin /Volumes/K66F/` .